src/array/cpu/spmm.cc

```diff
@@ -132,42 +154,42 @@ template void SpMMCsrHetero<kDLCPU, int32_t, 16>(
     const std::string& op, const std::string& reduce,
     const BcastOff& bcast, const std::vector<CSRMatrix>& csr,
     const std::vector<NDArray>& ufeat, const std::vector<NDArray>& efeat,
-    std::vector<NDArray> out, const std::vector<NDArray>& out_aux,
+    std::vector<NDArray> out, const std::vector<std::vector<NDArray>>& out_aux,
```

Comment:
Better to use `std::vector<NDArray> *` (`&` is also OK though I prefer the former) for in/out arguments.  Otherwise it will pass the vector by value.

Reply:
`&` will pass the vector by value?